### PR TITLE
Using correct SqlMetaData constructor for float column

### PR DESCRIPTION
**Pull request by snehabandla**
Reference:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.server.sqlmetadata.-ctor?view=dotnet-plat-ext-5.0
https://docs.microsoft.com/en-us/sql/t-sql/data-types/float-and-real-transact-sql?view=sql-server-ver15

### CODE REVIEW

File: src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs [hunk 1]
```diff
@@ -220,10 +232,26 @@ public override void Set(SqlDataRecord record, int ordinal, decimal value)
...
+            {
+                return 4;
+            }
+            else if (precision >= 24 && precision <= 53)
+            {
+                return 8;
+            }
+            else
+            {
+                throw new ArgumentOutOfRangeException(string.Format("Precision {0} must be between 1 & 53", precision));
```
Comment (brandonpollett) [inline src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs:251]: the error message should be in the resources file.

File: src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs [hunk 2]
```diff
@@ -220,10 +232,26 @@ public override void Set(SqlDataRecord record, int ordinal, decimal value)
     public class FloatColumn : Column<double>
     {
         public FloatColumn(string name, byte precision)
-            : base(name, SqlDbType.Float, false, precision, 0)
+            : base(name, SqlDbType.Float, false, GetLength(precision), precision, 0, 0, SqlCompareOptions.None, null)
         {
         }
 
+        private static long GetLength(byte precision)
```
Comment (brandonpollett) [inline src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs:239]: Any way to abstract this to where we're not copying it? Maybe a separate static class in this file?

**Review by brandonpollett — APPROVED** [review]

File: src/Microsoft.Health.SqlServer/Features/Schema/Model/ColumnUtilities.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+using System;
+
+namespace Microsoft.Health.SqlServer.Features.Schema.Model
+{
+    internal static class ColumnUtilities
+    {
+        internal static long GetLengthForFloatColumn(byte precision)
+        {
+            if (precision >= 1 && precision <= 23)
```
Comment (johnstairs) [inline src/Microsoft.Health.SqlServer/Features/Schema/Model/ColumnUtilities.cs:14]: In the docs, it looks like the upper bound is 24, not 23? https://docs.microsoft.com/en-us/sql/t-sql/data-types/float-and-real-transact-sql?view=sql-server-ver15#syntax

File: src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs [hunk 4]
```diff
@@ -220,7 +232,7 @@ public override void Set(SqlDataRecord record, int ordinal, decimal value)
     public class FloatColumn : Column<double>
     {
         public FloatColumn(string name, byte precision)
-            : base(name, SqlDbType.Float, false, precision, 0)
+            : base(name, SqlDbType.Float, false, ColumnUtilities.GetLengthForFloatColumn(precision), precision, 0, 0, SqlCompareOptions.None, null)
```
Comment (johnstairs) [inline src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs:235]: Kind of surprising, but it looks like this constructor, when called with float, throws ignores maxLength and precision. Bizarre api!

Comment (johnstairs) [reply src/Microsoft.Health.SqlServer/Features/Schema/Model/Column.cs:235]: I mean the SqlMetadata contructor :)

**Review by johnstairs — APPROVED** [review]

**Pull request merged**